openshift-ci/README.md

```diff
@@ -0,0 +1,214 @@
+namespace.  When you click on the job details from your pull request,
+you can see the name of the work namespace in the dashboard UI.  The
+name starts with `ci-op-`.  The images created goes into this
+temporary work namespace.  At the end of every image build, the [work
+namespace name has set as an environment variable][namespace] called
+`OPENSHIFT_BUILD_NAMESPACE`.  This environment variable is used to
+refer the image URLs inside the configuration files.
+
+## root
+
+As part of the CI pipeline, the first step is to create the `root`
+image.  In fact, `root` is a tag created for the pipeline image.  This
+image contains all the tools including but not limited to Go compiler,
+git, kubectl, oc, and Operator SDK.
+
+The `root` image tag is created using this Dockerfile:
+`openshift-ci/Dockerfile.tools`.
+
+## src
+
+This step clones the pull request branch with the latest changes.  The
+cloning is taking place inside a container created from the `root`
+image.  As a result of this step, an image named `src` is going to be
+created.
+
+In the CI configuration, there is a declaration like this:
+
+```
+canonical_go_repository: github.com/openshift/tektoncd-pipeline-operator
+```
+
+The above line ensures the cloned source code goes into the specified
+path: `$GOPATH/src/<canonical_go_repository>`.
+
+Note: If your pull request depends on any package installed through
+`yum install` inside the `root` image, those changes should be sent
+through a different PR and merged first.  As you can see from the
+above diagrams, the `src` image gets built after the `root` image,
+whereas the pull request branch gets merged in the `src` image.
+
+## bin
+
+This step runs the `build` Makefile target.  This step is taking place
+inside a container created from the `src` image created in the
+previous step.
+
+The `make build` produces an operator binary image available under
+`./out` directory.  Later, this binary is copied to
+`tektoncd-pipeline-operator` (see below).
+
+As a result of this step, a container image named `bin` is going to be
+created.
+
+## images
+
+Three container images are built as part of this job.  Before this
+step, a couple of base images are getting tagged from existing
+published images.  The first one is a CentOS 7 image referred to as
+`os` in the CI configuration.  The other one is the operator registry
+image which contains all the binaries to run a gRPC based registry.
+The operator image registry is referred to as `operator-registry` in
+the CI configuration.
+
+### tektoncd-pipeline-operator
+
+Thee CentOS 7 image (`os`) is going to use as the base image for
+creating `tektoncd-pipeline-operator` image.  The operator binary available
+inside `bin` container image is copied over here.  The Dockerfile used
+is available here: `openshift-ci/Dockerfile.deploy`
+
+The image produced can be pulled from here:
+`registry.svc.ci.openshift.org/${OPENSHIFT_BUILD_NAMESPACE}/stable:tektoncd-pipeline-operator`
```

Comment:
This depends on the `openshift/release` configuration. Is it gonna be `stable` for the name ?

Reply:
I will double check